components/Favorites/FavoriteItemsBar.tsx

```diff
@@ -0,0 +1,197 @@
+'use client'
+
+import { useEffect, useMemo, useState } from 'react'
+import Image from 'next/image'
+import Link from 'next/link'
+import { Badge, Spinner } from 'react-bootstrap'
+import api from '../../api/ApiHelper'
+import { CUSTOM_EVENTS } from '../../api/ApiTypes.d'
+import { convertTagToName, numberWithThousandsSeparators } from '../../utils/Formatter'
+import { getFavoriteItems } from '../../utils/FavoriteItemUtils'
+import FavoriteToggle from './FavoriteToggle'
+import SubscribeButton from '../SubscribeButton/SubscribeButton'
+import styles from './FavoriteItemsBar.module.css'
+
+const REFRESH_INTERVAL = 5 * 60 * 1000
+
+function FavoriteItemsBar() {
+    let [favorites, setFavorites] = useState<FavoriteItemEntry[]>(() => getFavoriteItems())
+    let [priceMovements, setPriceMovements] = useState<Record<string, ItemPriceMovement>>({})
+    let [isLoading, setIsLoading] = useState(false)
+    let [error, setError] = useState<string | null>(null)
+
+    useEffect(() => {
+        function load() {
+            setFavorites(getFavoriteItems())
+        }
+
+        load()
+
+        function onFavoritesUpdated(event?: Event) {
+            let detailFavorites = (event as CustomEvent)?.detail as FavoriteItemEntry[] | undefined
+            if (Array.isArray(detailFavorites)) {
+                setFavorites(detailFavorites)
+            } else {
+                load()
+            }
+        }
+
+        document.addEventListener(CUSTOM_EVENTS.FAVORITES_UPDATED, onFavoritesUpdated)
+        return () => {
+            document.removeEventListener(CUSTOM_EVENTS.FAVORITES_UPDATED, onFavoritesUpdated)
```

Comment:
please dont rely on custom events. If two components have to communicate with each other that are on different places in the tree, use a Provider